zerver/views/message_fetch.py

```diff
@@ -275,13 +274,16 @@ def messages_in_narrow_backend(
 
     if not is_search:
         # `add_narrow_conditions` adds the following columns only if narrow has search operands.
-        query = query.add_columns(topic_column_sa(), column("rendered_content", Text))
+        query = query.add_columns(
+            func.escape_html(topic_column_sa(), type_=Text).label("escaped_topic_name"),
+            column("rendered_content", Text),
+        )
```

Comment:
do you want me to add the same explanation here as well?

Reply:
I wonder if this is the wrong fix -- Like maybe the bug here is that `messages_in_narrow_backend` shouldn't be trying to access these columns if `!is_search` -- the escaped topic name is not expected to be if interest in that case.

(Somehow the message_fetch code path doesn't add these columns for `!is_search`, and the goal is for this code to work in a parallel fashion, so something feels awry).

Reply:
@timabbott so those columns are accessed because there are other use cases when we try to match the contents; such use case is when we use the `has` operator with the `link` operand.
(checkout the test case: `test_messages_in_narrow_for_non_search` in `zerver.tests.test_message_fetch.GetOldMessagesTest`.)